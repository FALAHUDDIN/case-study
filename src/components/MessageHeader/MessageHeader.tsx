import React from 'react';
import './MessageHeader.scss';
import {Box} from "@material-ui/core";
import IRootState from "../../types/IRootState";
import {connect} from "react-redux";
import {IMessage} from "../../types/IMessage";

interface IStateProps {
    messages?: IMessage[]
}

export interface Props extends IStateProps {
}

export const MessageHeader: React.FC<Props> = (props: Props) => {
    const unreadMessages = props.messages ? props.messages.filter(message => !message.isRead).length : 0
    return (
        <Box bgcolor="primary.main" color="primary.contrastText" p={2}>
            <strong>Messages</strong> {unreadMessages > 0 && <span className="MessageHeader__unreadMessages">{`(${unreadMessages} unread)`}</span>}
        </Box>
    )
}

const mapStateToProps = (state: IRootState): IStateProps => {
    return {
        messages: state.messages.messages
    }
}

export default connect(mapStateToProps, null)(MessageHeader)