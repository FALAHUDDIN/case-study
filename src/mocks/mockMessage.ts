import {IMessage} from "../types/IMessage";

export const MockMessage: IMessage = {
    id: 1,
    isRead: false,
    avatar: "person1.svg",
    sender: "Falah",
    message: "testmessage",
    received: "01/01/2020",
    subject: "testsubject",
    isArchived: false
}
export const MockMessageRead: IMessage = {
    ...MockMessage,
    isRead: true
}
export const MockMessageArchived: IMessage = {
    ...MockMessage,
    isArchived: true
}