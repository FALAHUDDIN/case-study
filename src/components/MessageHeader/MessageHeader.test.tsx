import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {MessageHeader, Props} from "./MessageHeader";
import {MockMessage, MockMessageRead} from "../../mocks/mockMessage";

describe('<MessageHeader/>', () => {
    let wrapper: ShallowWrapper;
    let props: Props = {
        messages: []
    };

    describe('without message', () => {
        beforeEach(() => {
            wrapper = shallow(<MessageHeader {...props} />)
        })

        it('displays no unread messages', () => {
            expect(wrapper.find('.MessageHeader__unreadMessages').length).toBe(0)
        })
    })

    describe('without read message', () => {
        beforeEach(() => {
            const mergedProps = {...props, messages: [MockMessageRead]}
            wrapper = shallow(<MessageHeader {...mergedProps} />)
        })

        it('displays no unread messages', () => {
            expect(wrapper.find('.MessageHeader__unreadMessages').length).toBe(0)
        })
    })

    describe('with unread message', () => {
        beforeEach(() => {
            const mergedProps = {...props, messages: [MockMessage, MockMessageRead]}
            wrapper = shallow(<MessageHeader {...mergedProps} />)
        })

        it('match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })

        it('displays unread messages text', () => {
            expect(wrapper.find('.MessageHeader__unreadMessages').length).toBe(1)
            expect(wrapper.find('.MessageHeader__unreadMessages').text()).toContain('1 unread')
        })
    })
});