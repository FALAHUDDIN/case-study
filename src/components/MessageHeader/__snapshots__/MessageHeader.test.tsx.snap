// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageHeader/> with unread message match snapshot 1`] = `
<Styled(MuiBox)
  bgcolor="primary.main"
  color="primary.contrastText"
  p={2}
>
  <strong>
    Messages
  </strong>
   
  <span
    className="MessageHeader__unreadMessages"
  >
    (1 unread)
  </span>
</Styled(MuiBox)>
`;
